Packet++/header/NflogLayer.h

```diff
@@ -101,23 +101,22 @@ namespace pcpp
 		/**
 		 * @return recordLen attribute in NflogTLVRawData
 		 */
-		size_t getTotalSize() const { return m_Data->recordLen; }
+		size_t getTotalSize() const
+		{
+			// as in https://github.com/the-tcpdump-group/libpcap/blob/766b607d60d8038087b49fc4cf433dac3dcdb49c/pcap-util.c#L371-L374
+			uint16_t size = m_Data->recordLen;
+			if (size % 4 != 0)
```

Comment:
yes this ALIGN macro will only work for a power of 2. 
ALIGN(11,2) => 12
ALIGN(11,4) => 12
ALIGN(11,8) => 16

I'm not sure we will to align on non power of 2 value no ?

You could add a static_assert inside the function to validate that the value is a power of 2.

I would say C++ function is more readable and easier to browse with GUI editor.

Reply:
@clementperon are you suggesting a simple `align()` method like:
```cpp
int align(int number, int alignment)
{
    int remainder = number % alignment;
    if (remainder == 0) 
    {
        return number;
    }
    else
    {
        return number + (alignment - remainder);
    }
}
```

Reply:
Yes, exactly and the align could avoid any if() if you use the formula from the kernel.

It will be better in term of performance

Reply:
@clementperon do you prefer a C++ function or a macro? It's not clear from the comments above... 
I personally think it doesn't really matters because the performance gain is relatively small compared to all the parsing these protocols do

Reply:
@seladb something like this.

Agree that performance gain will be small, but why not take it ?

```
constexpr bool is_powerof2(int v) {
    return v && ((v & (v - 1)) == 0);
}

// Only work for alignement with power of 2
int align(int number, int alignment)
{
    static_assert(is_powerof2(alignment))
    int mask = alignment - 1;
    return (number + mask) & ~mask;
}
```

Reply:
`the value of parameter "alignment" (declared at line 78) cannot be used as a constant`, so I have to remove the static assert. Please take a look.